e2e/suites/graphs.test.ts

```diff
@@ -0,0 +1,127 @@
+    await expect(page).toClick('button', { text: 'Save' });
+    await expect(page).toMatch('Saved successfully.');
+    await expect(page).toMatch('(view page)');
+  });
+
+  it('should display Bar chart graph in page with no more than a 3% difference', async () => {
+    const createdPageLink = await page.$eval(localSelectors.createdPageLink, el => (<HTMLLinkElement>el).href);
+    await page.goto(createdPageLink);
+    await page.waitFor(2000); // wait for the chart visualization animations to end
+
+    const chartContainer = await page.$('.recharts-responsive-container');
+    // @ts-ignore
+    const chartScreenshot = await chartContainer.screenshot();
+
+    expect(chartScreenshot).toMatchImageSnapshot({
+      failureThreshold: 0.03,
+      failureThresholdType: 'percent',
+      allowSizeMismatch: true,
+    });
+  });
+
+  it('should navigate back to the edit page and insert a Pie chart graph', async () => {
+    await page.goBack();
+    await page.click(localSelectors.pageContentsInput, {clickCount: 3});
+    await page.keyboard.press('Backspace');
+    await expect(page).toFill(localSelectors.pageContentsInput, '<Dataset />');
+    await page.type(localSelectors.pageContentsInput, graphs.pieChart);
+    await expect(page).toMatchElement('button', { text: 'Save' });
+    await expect(page).toClick('button', { text: 'Save' });
+    await expect(page).toMatch('Saved successfully.');
+    await expect(page).toMatch('(view page)');
+  });
+
+  it('should display Pie chart graph in page with no more than a 3% difference', async () => {
+    const createdPageLink = await page.$eval(localSelectors.createdPageLink, el => (<HTMLLinkElement>el).href);
+    await page.goto(createdPageLink);
+    await page.waitFor(2000);
+
+    const chartContainer = await page.$('.recharts-responsive-container');
+    // @ts-ignore
+    const chartScreenshot = await chartContainer.screenshot();
+
+    expect(chartScreenshot).toMatchImageSnapshot({
+      failureThreshold: 0.03,
+      failureThresholdType: 'percent',
+      allowSizeMismatch: true,
+    });
+  });
+
+  it('should navigate back to the edit page and insert a List chart graph', async () => {
+    await page.goBack();
+    await page.click(localSelectors.pageContentsInput, {clickCount: 3});
+    await page.keyboard.press('Backspace');
+    await expect(page).toFill(localSelectors.pageContentsInput, '<Dataset />');
+    await page.type(localSelectors.pageContentsInput, graphs.listChart);
+    await expect(page).toMatchElement('button', { text: 'Save' });
+    await expect(page).toClick('button', { text: 'Save' });
+    await expect(page).toMatch('Saved successfully.');
+    await expect(page).toMatch('(view page)');
+  });
+
+  it('should display List chart graph in page with no more than a 3% difference', async () => {
+    const createdPageLink = await page.$eval(localSelectors.createdPageLink, el => (<HTMLLinkElement>el).href);
+    await page.goto(createdPageLink);
```

Comment:
@elric-wamugu  i think we already discussed this on the nightmarejs version but can't we just do a click on the 'a' instead of an eval to get the url ? something like, same for the other cases where this technique is used.
```
toClick('a', { text: '(view page)'}) 
```

Reply:
@daneryl the main reason for using evaluate on html element here is to try and persist the page object instance (needed later for page element screenshot). The view page anchor element has the `target="_blank"` which opens the page on a new tab and thereby switching the browser target context. I've reworked the code to switch back the browser page context to the newly opened tab. 
This link provides more context into the issue https://github.com/puppeteer/puppeteer/issues/386